src/components/ConfirmLeavingPage/ConfirmUnsavedModal.test.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { DataTestIds } from 'test/dataTestIds';
+import { render } from 'test/render';
+
+import { ConfirmUnsavedModal } from './ConfirmUnsavedModal';
+
+describe('<ConfirmUnsavedModal>', () => {
+  it('should show when rendered', async () => {
+    render(<ConfirmUnsavedModal onLeavePage={jest.fn()} onStayOnPage={jest.fn()} />);
+
+    expect(await screen.findByTestId(DataTestIds.CONFIRM_UNSAVED_MODAL_HEADING)).toBeInTheDocument();
+  });
+
+  it('should call onLeavePage when leave page button clicked', async () => {
+    const onLeavePage = jest.fn();
+    render(<ConfirmUnsavedModal onLeavePage={onLeavePage} onStayOnPage={jest.fn()} />);
+
+    await screen.findByText('Leave page').then((button) => button.click());
+
+    expect(onLeavePage).toHaveBeenCalled();
+  });
+
+  it('should call onStayOnPage when stay on page button clicked', async () => {
+    const onStayOnPage = jest.fn();
+    render(<ConfirmUnsavedModal onLeavePage={jest.fn()} onStayOnPage={onStayOnPage} />);
+
+    await screen.findByText('Stay on page').then((button) => button.click());
+
+    expect(onStayOnPage).toHaveBeenCalled();
+  });
+
+  it('should call onStayOnPage when modal is dismissed (button)', async () => {
+    const onStayOnPage = jest.fn();
+    render(<ConfirmUnsavedModal onLeavePage={jest.fn()} onStayOnPage={onStayOnPage} />);
+
+    await screen.findByTestId('confirm-unsaved-modal-heading');
+    const closeButton = screen.getByRole('button', { name: 'Close' });
+
+    closeButton.click();
```

Comment:
We should use the `await user.click(closeButton)` returned from the render here.